### PR TITLE
Add team attribute to entities

### PR DESCRIPTION
**Pull request by Nico-Sanchez**
> [!Warning]
> Merge alongside https://github.com/lambdaclass/champions_of_mirra/pull/2245

## Motivation

We need a new team game mode.

## Summary of changes

- [Add new team attribute to Player proto](https://github.com/lambdaclass/mirra_backend/commit/8573959f3eec71c9cc9b230d0cbc19815aeb02b2)
- [Add new :team and :owner_team params to game entities](https://github.com/lambdaclass/mirra_backend/commit/d93e2a6cdfed659c2cdaa5eddfe81842538c7a12)
- [Refactor matchmaking launchers and assign team to players before game starts](https://github.com/lambdaclass/mirra_backend/commit/c6fb8e2973af12c06413da0773fdac8aaa78873c)
- [Update GameUpdater to receive players with teams and refactor the game initialization](https://github.com/lambdaclass/mirra_backend/commit/00f1cc4a85eb5df2f149ba709a49b7e93f841afe)
- [Use teams attribute to decide every game decision (damage, autoaim, etc.](https://github.com/lambdaclass/mirra_backend/commit/e5530ca5eefbca58348960e24983604fe5cda3d2)
- [Refactor dynamic obstacle initialization](https://github.com/lambdaclass/mirra_backend/commit/606e7790295d7025bb45c011876748c28a8cf7ee)
- [Add proto entry to send list of winners (team) instead of a single winner](https://github.com/lambdaclass/mirra_backend/commit/59800443fa2d1b33ddc0a1dc2ffbc7679b4fdc58)
- [Update winning condition due to new teams concept](https://github.com/lambdaclass/mirra_backend/commit/6df7ac96250fe03eff44d26fb47a0c1159a9c5a9)
- [Move repeated code in matchmaking queues to Utils module & add bot name generator](https://github.com/lambdaclass/mirra_backend/commit/caaa0fa61d56eda3bf2d780705a864482a4be56d) 
- [Update game tracker functions due to changes in code (teams)](https://github.com/lambdaclass/mirra_backend/commit/faec651383e48530177b4d5340ea3c9d429e30da)
- [Update arena version due to new and changed messages in Protobuf](https://github.com/lambdaclass/mirra_backend/commit/29ddd557f716871fcef179ffc0340f1c1d871969)

## How to test it?

Play the game with the front [branch](https://github.com/lambdaclass/champions_of_mirra/issues/2242). Everything should work as usual.
Pay special attention to collisions, damage, effects, pools, projectiles, etc. Everything should affect the proper entities.

## Checklist
- [x] Tested the changes locally.
- [x] Reviewed the changes on GitHub, line by line.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Comment by tvillegas98** [comment]
This PR also closes #960

File: apps/arena/lib/arena/matchmaking/game_launcher.ex [hunk 1]
```diff
@@ -51,16 +29,24 @@ defmodule Arena.Matchmaking.GameLauncher do
   def handle_call({:join, client_id, character_name, player_name}, {from_pid, _}, %{clients: clients} = state) do
     batch_start_at = maybe_make_batch_start_at(state.clients, state.batch_start_at)
 
+    client = %{
+      client_id: client_id,
+      character_name: character_name,
+      name: player_name,
+      from_pid: from_pid,
+      type: :human
```
Comment (tvillegas98) [inline apps/arena/lib/arena/matchmaking/game_launcher.ex:37]: Is there a reason for this `type` parameter? just wondering

**Review by tvillegas98 — DISMISSED** [review]
LGTM just left a question

File: apps/arena/lib/arena/matchmaking/game_launcher.ex (hunk 1 above)
Comment (Nico-Sanchez) [reply apps/arena/lib/arena/matchmaking/game_launcher.ex:37]: It's used by the game_updater genserver to save human and bot ids in different lists.
See game_updater.ex:80-84

**Review by tvillegas98 — APPROVED** [review]

**Review by manucamejo — CHANGES_REQUESTED** [review]
Hey @Nico-Sanchez I'm getting an error when the match ends in Deathmatch mode.
```elixir
** (BadMapError) expected a map, got: 12
    (elixir 1.16.2) lib/map.ex:682: Map.update(12, :category, nil, #Function<8.2673582/1 in Arena.GameUpdater.complete_entity/2>)
    (arena 0.11.1) lib/arena/game_updater.ex:812: Arena.GameUpdater.complete_entity/2
    (arena 0.11.1) lib/arena/game_updater.ex:803: anonymous fn/3 in Arena.GameUpdater.complete_entities/2
    (stdlib 5.2.1) maps.erl:416: :maps.fold_1/4
    (arena 0.11.1) lib/arena/game_updater.ex:786: Arena.GameUpdater.broadcast_game_ended/2
    (arena 0.11.1) lib/arena/game_updater.ex:353: Arena.GameUpdater.handle_info/2
    (stdlib 5.2.1) gen_server.erl:1095: :gen_server.try_handle_info/3
    (stdlib 5.2.1) gen_server.erl:1183: :gen_server.handle_msg/6
    (stdlib 5.2.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
Last message: :deathmatch_end_game_check
```
Can you solve it?

**Review by manucamejo — APPROVED** [review]

**Review by tvillegas98 — APPROVED** [review]

**Pull request merged**